lib/languageFeatures/documentHightlightHandler.ts

```diff
@@ -1,25 +1,67 @@
-    const candidate = candidates.filter(candidate => implementsIHasLexerToken(candidate))[0] as IHasLexerToken;
-    if (typeof candidate === 'undefined') {
-        return null;
+
+    const candidates = findObjectFromPosition(linter, params.position)
+        .filter(candidate => {
+            // Only use Objects that have a lexer Token (either directly or as a reference Token)
+            // Also check if the cursor is actually in that token itself, or just in that objects range.
+            if (implementsIHasLexerToken(candidate)) {
+                if (candidate.lexerToken.range.start.i <= startI && startI <= candidate.lexerToken.range.end.i) {
+                    return true;
+                }
+            } else if (implementsIHasReferenceToken(candidate)) {
+                if (candidate.referenceToken.range.start.i <= startI && startI <= candidate.referenceToken.range.end.i) {
+                    return true;
+                }
+            }
+            return false;
+
+        }) as (ObjectBase & (IHasLexerToken | IHasReferenceToken))[]
+
+    const candidate = candidates[0];
+    if (candidate === undefined) {
+        return [];
+    }
+    const highlights: DocumentHighlight[] = [];
+    if (implementsIHasLexerToken(candidate)) {
+        const name = candidate.lexerToken.getLText();
+        for (const object of linter.file.objectList) {
+            if (implementsIHasLexerToken(object) && object.lexerToken.getLText() === name) {
+                highlights.push({
+                    range: object.lexerToken.range,
+                    kind: object instanceof OWrite ? DocumentHighlightKind.Write : DocumentHighlightKind.Read
+                });
+            } else if (implementsIHasReferenceToken(object) && object.referenceToken.getLText() === name) {
+                highlights.push({
+                    range: object.referenceToken.range,
+                    kind: object instanceof OWrite ? DocumentHighlightKind.Write : DocumentHighlightKind.Read
+                });
+            }
+        }
+
+    }
+    if (implementsIHasReferenceToken(candidate)) {
+        const name = candidate.referenceToken.getLText();
+        for (const object of linter.file.objectList) {
+            if (implementsIHasLexerToken(object) && object.lexerToken.getLText() === name) {
+                highlights.push({
+                    range: object.lexerToken.range,
+                    kind: object instanceof OWrite ? DocumentHighlightKind.Write : DocumentHighlightKind.Read
+                });
+            } else if (implementsIHasReferenceToken(object) && object.referenceToken.getLText() === name) {
+                highlights.push({
+                    range: object.referenceToken.range,
+                    kind: object instanceof OWrite ? DocumentHighlightKind.Write : DocumentHighlightKind.Read
+                });
+            }
+        }
```

Comment:
That's the same code as lines 33 to 45.

Reply:
No its not, name is different. But i can DRY it with extra function